strawberryfields/parameters.py

```diff
@@ -90,17 +90,20 @@
 ~~~~~~~~~~~~
 
 """
-
 import numbers
 
+import mock
 import numpy as np
-import tensorflow as tf
 
 from .program import (RegRef, RegRefTransform)
 
 
-# supported TF classes
-_tf_classes = (tf.Tensor, tf.Variable)
+try:
+    import tensorflow as tf
+    _tf_classes = (tf.Tensor, tf.Variable)
+except ImportError:
+    tf = mock.MagicMock()
```

Comment:
What's the point of the mock?

Reply:
There are (module level) calls to TensorFlow classes, types, and functions, which are only used if the user is using the `'tf'` backend. If the user is using the `'tf'` backend, the `ImportError` exception in `tfbackend/__init__.py` will be raised before the `parameter.py` file is used.

This branch takes care of the case where TensorFlow is not installed, and the user wants to use the `'fock'` or `'gaussian'` backends. The alternative to mocking would be a significant refactor to change the logic required; since the `Parameter` class still needs to be refactor/replaced anyway this cycle, it doesn't make sense to do this now.